posthog/api/insight.py

```diff
@@ -172,9 +186,27 @@ def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> Insight:
 
         # Manual tag creation since this create method doesn't call super()
         self._attempt_set_tags(tags, insight)
+
+        name_for_logged_activity = (
+            insight.name if insight.name is not None and len(insight.name) > 0 else insight.derived_name
+        )
```

Comment:
Nit: I'm not python enough to know if this is the best way to do this... but it looks long

Reply:
If I'm understanding correctly:

`if insight.name is not None and len(insight.name) > 0` is same as `if insight.name` ? (`[]` and `''` and any other thing that should be falsy is falsy, unlike js) xD

Reply:
🤯 

But programming languages are _supposed_ to be confusing! :)